src/main/app/src/app/zaken/zaak-koppelen-dialog/zaak-koppelen-dialog.component.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * SPDX-FileCopyrightText: 2022 Atos
+ * SPDX-License-Identifier: EUPL-1.2+
+ */
+
+import {Component, Inject, OnInit} from '@angular/core';
+import {MAT_DIALOG_DATA, MatDialogRef} from '@angular/material/dialog';
+import {ZakenService} from '../zaken.service';
+import {MaterialFormBuilderService} from '../../shared/material-form-builder/material-form-builder.service';
+import {AbstractFormField} from '../../shared/material-form-builder/model/abstract-form-field';
+import {Zaak} from '../model/zaak';
+import {ZaakKoppelGegevens} from '../model/zaak-koppel-gegevens';
+import {SelectFormFieldBuilder} from '../../shared/material-form-builder/form-components/select/select-form-field-builder';
+import {Validators} from '@angular/forms';
+import {UtilService} from '../../core/service/util.service';
+import {GerelateerdeZaak} from '../model/gerelateerde-zaak';
+
+@Component({
+    templateUrl: 'zaak-koppelen-dialog.component.html'
+})
+export class ZaakKoppelenDialogComponent implements OnInit {
+
+    koppelSelectFormField: AbstractFormField;
+    bronZaak: Zaak;
+    teKoppelenZaak: Zaak;
+    loading: boolean;
+    koppelKeuzes: { label: string, value: string }[] = [];
+
+    constructor(
+        public dialogRef: MatDialogRef<ZaakKoppelenDialogComponent>,
+        @Inject(MAT_DIALOG_DATA) public data: ZaakKoppelGegevens,
+        private mfbService: MaterialFormBuilderService,
+        private zakenService: ZakenService,
+        private utilService: UtilService) {
+    }
+
+    close(): void {
+        this.zakenService.addTeKoppelenZaak(this.bronZaak);
+        this.dialogRef.close();
+    }
+
+    ngOnInit(): void {
+        this.loading = true;
+        this.zakenService.readZaak(this.data.bronZaakUuid).subscribe(bronZaak => {
+            this.bronZaak = bronZaak;
+            this.zakenService.readZaakByID(this.data.gerelateerdeZaak.identificatie).subscribe(teKoppelenZaak => {
+                this.teKoppelenZaak = teKoppelenZaak;
```

Comment:
Je leest nu de te Koppelenzaak na her lezen van de bron zaak maar je zou ze tegelijk kuinnen lezen. Daar is een RxJS operator voor. Ik weet niet meer precies hoe dat werk. Maar neem de tijd om daar eens in te duiken en te leren. Er is bijvoorbeeld een goed pluralsight over.